src/main/java/com/google/cloud/tools/gradle/appengine/flexible/FlexibleDeployTargetResolver.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018 Google Inc. All Right Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.google.cloud.tools.gradle.appengine.flexible;
+
+import com.google.cloud.tools.gradle.appengine.core.DeployTargetResolver;
+import org.gradle.api.GradleException;
+
+public class FlexibleDeployTargetResolver implements DeployTargetResolver {
+
+  @Override
+  public String getProject(String configString) {
+    if (configString == null
+        || configString.trim().isEmpty()
+        || configString.equals(APPENGINE_CONFIG)) {
+      throw new GradleException(
+          "project must be defined or configured to read from system state\n"
```

Comment:
Should probably capitalize the start of each sentence here (same for all the other places you throw an exception).